sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -759,22 +761,29 @@ class Analyzer(
           .getOrElse(i)
 
       case desc @ DescribeTable(u: UnresolvedV2Relation, _) =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName)
-            .map(rel => desc.copy(table = rel))
-            .getOrElse(desc)
+        resolveV2Relation(u).map(rel => desc.copy(table = rel)).getOrElse(desc)
 
       case alter @ AlterTable(_, _, u: UnresolvedV2Relation, _) =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName)
-            .map(rel => alter.copy(table = rel))
-            .getOrElse(alter)
+        resolveV2Relation(u).map(rel => alter.copy(table = rel)).getOrElse(alter)
 
       case show @ ShowTableProperties(u: UnresolvedV2Relation, _) =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName)
-          .map(rel => show.copy(table = rel))
-          .getOrElse(show)
+        resolveV2Relation(u).map(rel => show.copy(table = rel)).getOrElse(show)
 
       case u: UnresolvedV2Relation =>
-        CatalogV2Util.loadRelation(u.catalog, u.tableName).getOrElse(u)
+        resolveV2Relation(u).getOrElse(u)
+    }
+
+    private def resolveV2Relation(unresolved: UnresolvedV2Relation) : Option[NamedRelation] = {
+      val maybeTempView = unresolved.originalNameParts match {
+        case Seq(part) => v1SessionCatalog.lookupTempView(part)
+        case _ => None
+      }
+      if (maybeTempView.isDefined) {
+        unresolved.failAnalysis(
+          s"Invalid command: '${unresolved.originalNameParts.quoted}' is a view not a table.")
```

Comment:
I left this to the same message used in checkAnalysis to minimize the changes in tests. Please let me know if this neds to updated to more descriptive message such as 'A temp view 't' cannot be handled by V2 commands'